lib/LLVMIRCodeGen/BundleSaver.cpp

```diff
@@ -84,6 +138,160 @@ void BundleSaver::saveWeights(llvm::StringRef weightsFileName) {
   weightsFile.close();
 }
 
+void BundleSaver::saveHeader(llvm::StringRef headerFileName) {
+  std::error_code EC;
+  llvm::raw_fd_ostream headerFile(headerFileName, EC,
+                                  llvm::sys::fs::OpenFlags::F_Text);
+  CHECK(!EC) << "Could not open header file!";
+  auto bundleName = irgen_->getBundleName();
+  auto bundleNameUpper = llvm::StringRef(bundleName).upper();
+  auto headerDefine = "_GLOW_BUNDLE_" + bundleNameUpper + "_H";
+  auto commonDefine = "_GLOW_BUNDLE_COMMON_DEFS";
+  auto constMemSize = irgen_->getAllocationsInfo().constantWeightVarsMemSize_;
+  auto mutableMemSize = irgen_->getAllocationsInfo().mutableWeightVarsMemSize_;
+  auto activationsMemSize = irgen_->getAllocationsInfo().activationsMemSize_;
+  auto memAlignSize = TensorAlignment;
+  auto totMemSize = constMemSize + mutableMemSize + activationsMemSize;
+
+  // Print file header.
+  headerFile << "// Bundle API header file\n"
+             << "// Auto-generated file. Do not edit!\n"
+             << "#ifndef " << headerDefine << "\n"
+             << "#define " << headerDefine << "\n\n"
+             << "#include <stdint.h>\n\n";
+
+  // Print common bundle definitions.
+  headerFile
+      << "// ---------------------------------------------------------------\n"
+      << "//                       Common definitions                       \n"
+      << "// ---------------------------------------------------------------\n"
+      << "#ifndef " << commonDefine << "\n"
+      << "#define " << commonDefine << "\n\n";
+  if (bundleApi == BundleApiType::Dynamic) {
+    headerFile
+        << "// Type describing a symbol table entry of a generated bundle.\n"
```

Comment:
You may consider using [raw string literals](https://en.cppreference.com/w/cpp/language/string_literal) for these multi-line strings and check if the code looks better this way.

Reply:
A couple of considerations here: 
- There is a small danger that these definitions may get out of sync with the actual format of the bundle, should it change. But this will be detected pretty quickly as tests will most likely start failing.
- Since these definitions here are not dependent on any dynamic information, they could be defined in a self-contained header file, which is part of glow and would be just included by the generated header.

Reply:
I have changed the common defines to use raw string literals (it is indeed clearer).

Regarding your 2 points:
- The bundle format has not changed for quite some time now. If it does the tests should capture that (lenet_mnist and resnet50 bundle tests)
- The common definitions could be in principle put in a separate header file. But I like the idea having the bundle artifacts (produced by the compiler) self-sufficient (with no other external dependencies). The common definitions are also guarded with an #ifdef preprocessor guard to avoid multiple inclusions when multiple bundles are used together (I`ve tested this). Having a separate stand-alone file I think will be more prone to getting misaligned with the true bundle format.